agent/structs/structs.go

```diff
@@ -513,6 +513,9 @@ type ServiceSpecificRequest struct {
 	// Connect if true will only search for Connect-compatible services.
 	Connect bool
 
+	// Ingress if true will only search for Ingress gateways for the given service.
+	Ingress bool
```

Comment:
Does this need a corresponding `api` change?

Reply:
Yes, a new query param in the health/service/:serviceid endpoint should go along with this

Reply:
@kyhavlov and I discussed this offline a bit, and realized that it might not be as simple as adding a new query param, since the similar `Connect bool` field has an entire endpoint associated to it, `/health/connect/:serviceid`.

We might punt on figuring this out in this PR and leave a TODO action for ourselves to come back and figure this out.